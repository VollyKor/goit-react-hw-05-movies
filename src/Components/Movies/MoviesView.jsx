export default function MoviesView() {
  return (
    <ul>
      <li>MoviesView</li>
      <li>2</li>
      <li>3</li>
    </ul>
  );
}
