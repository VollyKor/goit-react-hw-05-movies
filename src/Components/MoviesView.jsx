import Gallery from 'components/Gallery/Gallery';

const MoviesView = ({ data }) => data && <Gallery data={data.results} />;

export default MoviesView;
