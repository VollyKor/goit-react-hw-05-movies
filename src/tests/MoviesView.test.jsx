import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';

import MoviesView from 'components/MoviesView';

import { MoviesViewData } from './mock/data';

describe('Test Component MoviesView', () => {
  it('renders correctle', () => {
    const scr = render(
      <MemoryRouter>
        <MoviesView data={MoviesViewData} />
      </MemoryRouter>,
    );

    expect(scr.container).toContainHTML('<body>');
  });

  it('render correctly', () => {
    const scr = render(
      <MemoryRouter>
        <MoviesView data={MoviesViewData} />
      </MemoryRouter>,
    );

    expect(scr).toMatchSnapshot();
  });
});
