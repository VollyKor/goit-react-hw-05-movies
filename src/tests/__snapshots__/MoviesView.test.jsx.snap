// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Component MoviesView render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="list"
      >
        <li
          class="item"
        >
          <a
            class="link"
            href="/movies/shrek-808"
          >
            <div
              class="thumb"
            >
              <img
                alt="Shrek"
                class="img"
                src="https://image.tmdb.org/t/p/w500/2l5UHZBcp9cx1PwKLdisJ0gV9jB.jpg"
              />
              <p
                class="title"
              >
                Shrek
              </p>
            </div>
          </a>
        </li>
        <li
          class="item"
        >
          <a
            class="link"
            href="/movies/shrek2-8082"
          >
            <div
              class="thumb"
            >
              <img
                alt="Shrek2"
                class="img"
                src="https://image.tmdb.org/t/p/w500/2l5UHZBcp9cx1PwKLdisJ0gV9jB2.jpg"
              />
              <p
                class="title"
              >
                Shrek2
              </p>
            </div>
          </a>
        </li>
        <li
          class="item"
        >
          <a
            class="link"
            href="/movies/shrek3-80212138"
          >
            <div
              class="thumb"
            >
              <img
                alt="Shrek3"
                class="img"
                src="https://image.tmdb.org/t/p/w500/2l5UHZBcp9cx1PwKLdisJ0gV9jB.jpg3"
              />
              <p
                class="title"
              >
                Shrek3
              </p>
            </div>
          </a>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="list"
    >
      <li
        class="item"
      >
        <a
          class="link"
          href="/movies/shrek-808"
        >
          <div
            class="thumb"
          >
            <img
              alt="Shrek"
              class="img"
              src="https://image.tmdb.org/t/p/w500/2l5UHZBcp9cx1PwKLdisJ0gV9jB.jpg"
            />
            <p
              class="title"
            >
              Shrek
            </p>
          </div>
        </a>
      </li>
      <li
        class="item"
      >
        <a
          class="link"
          href="/movies/shrek2-8082"
        >
          <div
            class="thumb"
          >
            <img
              alt="Shrek2"
              class="img"
              src="https://image.tmdb.org/t/p/w500/2l5UHZBcp9cx1PwKLdisJ0gV9jB2.jpg"
            />
            <p
              class="title"
            >
              Shrek2
            </p>
          </div>
        </a>
      </li>
      <li
        class="item"
      >
        <a
          class="link"
          href="/movies/shrek3-80212138"
        >
          <div
            class="thumb"
          >
            <img
              alt="Shrek3"
              class="img"
              src="https://image.tmdb.org/t/p/w500/2l5UHZBcp9cx1PwKLdisJ0gV9jB.jpg3"
            />
            <p
              class="title"
            >
              Shrek3
            </p>
          </div>
        </a>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
